### DISCUSSION TITLE
Error: Invalid command block, it's not registered.

### DISCUSSION THREAD

**Discussion by samucati** [umijs/umi#10405] [General]
hello, I'm trying to add a new block and I get the following error:

`@samucati ➜ /workspaces/ant-design-pro (main) $ umi block add https://github.com/samucati/pro-blocks/tree/master/UserRegister
bash: umi: command not found
@samucati ➜ /workspaces/ant-design-pro (main) $ **npm install -g umi**
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated @floating-ui/react-dom-interactions@0.3.1: Package renamed to @floating-ui/react

added 971 packages in 21s

210 packages are looking for funding
  run `npm fund` for details
@samucati ➜ /workspaces/ant-design-pro (main) $ **umi block add** https://github.com/samucati/pro-blocks/tree/master/UserRegister
**fatal - Error: Invalid command block, it's not registered.**
    at Service.run (/usr/local/share/nvm/versions/node/v19.5.0/lib/node_modules/umi/node_modules/@umijs/core/dist/service/service.js:257:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Service.run2 (/usr/local/share/nvm/versions/node/v19.5.0/lib/node_modules/umi/dist/service/service.js:61:12)
    at async Object.run (/usr/local/share/nvm/versions/node/v19.5.0/lib/node_modules/umi/dist/cli/cli.js:55:7)
fatal - A complete log of this run can be found in:
fatal - /workspaces/ant-design-pro/node_modules/.cache/logger/umi.log
fatal - Consider reporting a GitHub issue on https://github.com/umijs/umi/issues
@samucati ➜ /workspaces/ant-design-pro (main) $ `

What can it be?

**Comment by fz6m** [comment]
The latest version is `umi 4` , command `umi block` is a `umi 2` feature, see https://v2.umijs.org/guide/block.html

You can try: 

```bash
  npm i -D umi@^2
  npx umi block add ...
```
